vxquery-core/src/main/java/org/apache/vxquery/runtime/functions/json/AbstractLibjnAccumulateScalarEvaluator.java

```diff
@@ -0,0 +1,170 @@
+        ab = new ArrayBuilder();
+        tvps = new LinkedHashMap<>();
+    }
+
+    @Override
+    protected void evaluate(TaggedValuePointable[] args, IPointable result) throws SystemException {
+        TaggedValuePointable arg = args[0];
+        if (!(arg.getTag() == ValueTag.SEQUENCE_TAG || arg.getTag() == ValueTag.OBJECT_TAG)) {
+            throw new SystemException(ErrorCode.FORG0006);
+        }
+        TaggedValuePointable tempTvp = ppool.takeOne(TaggedValuePointable.class);
+        try {
+            abvs.reset();
+            ob.reset(abvs);
+            tvps.clear();
+            if (arg.getTag() == ValueTag.SEQUENCE_TAG) {
+                arg.getValue(sp);
+                for (int i = 0; i < sp.getEntryCount(); ++i) {
+                    op = (ObjectPointable) ObjectPointable.FACTORY.createPointable();
+                    sp.getEntry(i, tempTvp);
+                    if (tempTvp.getTag() == ValueTag.OBJECT_TAG) {
+                        tempTvp.getValue(op);
+                        addPairs(tempTvp);
+                    }
+                }
+            } else if (arg.getTag() == ValueTag.OBJECT_TAG) {
+                op = (ObjectPointable) ObjectPointable.FACTORY.createPointable();
+                arg.getValue(op);
+                addPairs(tempTvp);
+            }
+            for (TaggedValuePointable key1 : tvps.keySet()) {
+                key1.getValue(stringKey);
+                values = tvps.get(key1);
+                if (values.size() > 1) {
+                    FunctionHelper.removeDuplicates(values);
+                    if (values.size() > 1) {
+                        abvs1.reset();
+                        ab.reset(abvs1);
+                        for (TaggedValuePointable pointable : values) {
+                            ab.addItem(pointable);
+                        }
+                        ab.finish();
+                        ob.addItem(stringKey, abvs1);
+                    } else {
+                        ob.addItem(stringKey, values.get(0));
+                    }
+                } else {
+                    addKeyValue(stringKey, values.get(0));
+                }
+            }
+            ob.finish();
+            result.set(abvs);
+        } catch (IOException e) {
+            throw new SystemException(ErrorCode.SYSE0001, e);
+        } finally {
+            ppool.giveBack(tempTvp);
+            for (TaggedValuePointable key1 : tvps.keySet()) {
+                for (TaggedValuePointable value1 : tvps.get(key1)) {
+                    ppool.giveBack(value1);
+                }
+                ppool.giveBack(key1);
+            }
+        }
+    }
+
+    protected abstract void addKeyValue(UTF8StringPointable stringKey, TaggedValuePointable taggedValuePointable) throws IOException;
+
+    private void addPair(TaggedValuePointable tempTvp) throws IOException, SystemException {
+        TaggedValuePointable tvp = isDuplicateKeys(tempTvp, tvps.keySet());
```

Comment:
It is not a simple check because we have to use the byte array to see if the keys are equal. And the key that is added the Map could only be used to retrieve the value/s and not any `TaggedValuePointable` which has similar byte array :/